apps/theming-designer/src/components/FabricSlotWidget.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { IThemeRules, FabricSlots, IThemeSlotRule } from 'office-ui-fabric-react/lib/ThemeGenerator';
+import { IColor } from 'office-ui-fabric-react/lib/Color';
+import { Stack } from 'office-ui-fabric-react/lib/Stack';
+import { mergeStyles } from '@uifabric/merge-styles';
+import { ColorPicker } from 'office-ui-fabric-react/lib/ColorPicker';
+import { Callout } from 'office-ui-fabric-react/lib/Callout';
+
+export interface IFabricSlotWidgetProps {
+  slot: FabricSlots;
+  onFabricPaletteColorChange: (newColor: IColor | undefined, fabricSlot: FabricSlots) => void;
+  themeRules?: IThemeRules;
+}
+
+export interface IFabricSlotWidgetState {
+  isColorPickerVisible: boolean;
+  colorPickerElement: HTMLElement | null;
+  slotRule: IThemeSlotRule | null;
+}
+
+const slotClassName = mergeStyles({
+  display: 'flex',
+  alignItems: 'center',
+  overflow: 'auto'
+});
+
+const fabricPaletteColorBox = mergeStyles({
+  width: 15,
+  height: 15,
+  display: 'inline-block',
+  left: 5,
+  top: 5,
+  border: '1px solid black',
+  flexShrink: 0
+});
+
+export class FabricSlotWidget extends React.Component<IFabricSlotWidgetProps, IFabricSlotWidgetState> {
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      isColorPickerVisible: false,
+      colorPickerElement: null,
+      slotRule: props.themeRules![FabricSlots[this.props.slot]]
```

Comment:
You can probably pull this data directly from props and not require a state property for it judging by current code.